pkg/aws/client_test.go

```diff
@@ -38,6 +41,57 @@ var _ = Describe("Client", func() {
 		mockCtrl.Finish()
 	})
 
+	Context("ValidateCFUserCredentials", func() {
+		var (
+			CFsecretID      = "sut"
+			IAMAccessKey    = "sut"
+			oddIAMAccessKey = "longtestkey"
+		)
+		Context("when creds are OK and matches", func() {
+			BeforeEach(func() {
+				mockCfAPI.EXPECT().DescribeStackResource(gomock.Any()).Return(&cloudformation.DescribeStackResourceOutput{
+					StackResourceDetail: &cloudformation.StackResourceDetail{
+						PhysicalResourceId: &CFsecretID,
+					},
+				}, nil)
+
+				mockIamAPI.EXPECT().ListAccessKeys(gomock.Any()).Return(&iam.ListAccessKeysOutput{
+					AccessKeyMetadata: []*iam.AccessKeyMetadata{
+						{
+							AccessKeyId: &IAMAccessKey,
+						},
+					},
+				}, nil)
+			})
+			It("should return nil", func() {
+				err := client.ValidateCFUserCredentials()
+				Expect(err).To(BeNil())
+			})
+		})
+
+		Context("when the creds doesn't match", func() {
+			BeforeEach(func() {
+				mockCfAPI.EXPECT().DescribeStackResource(gomock.Any()).Return(&cloudformation.DescribeStackResourceOutput{
+					StackResourceDetail: &cloudformation.StackResourceDetail{
+						PhysicalResourceId: &CFsecretID,
+					},
+				}, nil)
+
+				mockIamAPI.EXPECT().ListAccessKeys(gomock.Any()).Return(&iam.ListAccessKeysOutput{
+					AccessKeyMetadata: []*iam.AccessKeyMetadata{
+						{
+							AccessKeyId: &oddIAMAccessKey,
+						},
+					},
+				}, nil)
+			})
+			It("should return err", func() {
+				err := client.ValidateCFUserCredentials()
+				Expect(err.Error()).Should(ContainSubstring("you can recreate the CloudFormation stack with `moactl init --delete-stack`"))
```

Comment:
IMO the error should not be such a big section of the error itself, maybe reorder it so the front of the error will a short description of it you can search for and the end will include the dynamic parts:

`
Invalid CloudFormation stack credentials: AdminUser %s is not valid, to recreate run "moactl init --delete-stack && moactl init"
`

this will make the error querying:
```suggestion
				Expect(err.Error()).Should(ContainSubstring("Invalid CloudFormation stack credentials"))
```

which is what the user will be searching anyway online if it was me